src/josepy/jwa.py

```diff
@@ -181,7 +181,7 @@ def _sign(self, key: ec.EllipticCurvePrivateKey, msg: bytes) -> bytes:
         try:
             if new_api:
                 return key.sign(msg, ec.ECDSA(self.hash))
-            signer = key.signer(ec.ECDSA(self.hash))
+            signer = key.signer(ec.ECDSA(self.hash))  # type: ignore[attr-defined]
```

Comment:
Do you know what's going on with the `ignore` lines here? It seems like these attributes exist. What should ideally be "fixed" so these ignores are no longer required in your opinion?

Reply:
Yes basically this code relates to old versions of cryptography where `sign` function was not existing. `signer` and `verifier` methods have been deprecated since cryptography 2.1.

It seems that the `types-cryptography` stub package has been proactive and these functions have already been removed, certainly related to the final deprecation warning made by cryptography that 37.0.0 will remove these functions, in order to push developers to stop using them.

The simplest fix I think it to declare dependency on `cryptography>=1.5` instead of `cryptography>=0.8`, and stop using this retro-compatible branch.

Reply:
Works for me ! Cryptography 1.5 has been added 5 years ago, I think it is fine in term of backward compatibility.

Reply:
Dependencies and constraints updated.